charmcraft/linters.py

```diff
@@ -376,6 +377,106 @@ def run(self, basedir: pathlib.Path) -> str:
         return self.Result.OK
 
 
+class NamingConventions(Linter):
+    """Check that charm follows naming conventions.
+
+    More information can be found at https://juju.is/docs/sdk/styleguide#heading--naming.
+    """
+
+    name = "naming-conventions"
+    url = "https://juju.is/docs/sdk/styleguide#heading--naming"
+
+    exception_result = LintResult.WARNING
+
+    def __init__(self):
+        self.text = ""
+
+    @staticmethod
+    def check_naming_convention(names: typing.Iterable[str], scope: str) -> str | None:
+        """Check adherance to naming convention.
+
+        :returns: string with warning if present, otherwise None
+        """
+        snake_keys = [key for key in names if "_" in key]
+
+        if snake_keys:
+            hyphen_keys = [key for key in names if "-" in key]
+
+            if hyphen_keys:
+                return (
+                    f"Some {scope} ({', '.join(snake_keys)}) are in snake case, "
+                    f"while others  ({', '.join(hyphen_keys)}) are with hyphens."
+                )
+            else:
+                return (
+                    f"Some {scope} ({', '.join(snake_keys)}) are using "
+                    "snake case naming convention."
+                )
+
+        return None
+
+    @staticmethod
+    def _config_options_check(config_file: pathlib.Path) -> list[str]:
+        # This is safe as the compliance with YAML is done in the JujuConfig linter
+        warnings = []
+
+        if not config_file.exists():
+            return warnings
+
+        with config_file.open("rt", encoding="utf8") as fh:
+            options = yaml.safe_load(fh).get("options")
```

Comment:
```suggestion
            options = yaml.safe_load(fh).get("options", {})
```

The user will get an error from the ConfigLinter before this runs, but this will still run so we should ensure it can handle invalid files.

Likewise both here and in `_actions_check`, please handle the cases where the file is empty (`safe_load` returns `None`) and where it's not valid YAML. We don't need to produce warnings again, we can just silently return. (And please test for these cases)

Reply:
I believe in the actions this is already checked. Anyhow, I have added a test case to make sure that this was indeed true.